test/services/notices/setup/determine-notice-type.service.test.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+
+const { describe, it } = (exports.lab = Lab.script())
+const { expect } = Code
+
+// Thing under test
+const DetermineNoticeTypeService = require('../../../../app/services/notices/setup/determine-notice-type.service.js')
+
+describe('Notices - Setup - Initiate Session service', () => {
+  describe('when called', () => {
+    describe('when the "notificationType" is "invitations"', () => {
+      it('creates a new session record', () => {
+        const result = DetermineNoticeTypeService.go('invitations')
+
+        expect(result).to.equal({
+          journey: 'invitations',
+          name: 'Returns: invitation',
+          notificationType: 'Returns invitation',
+          redirectPath: 'returns-period',
+          referenceCode: result.referenceCode, // randomly generated
+          subType: 'returnInvitation'
+        })
+      })
+
+      describe('the "referenceCode" property', () => {
+        it('returns a reference code for "invitations" notifications', () => {
+          const result = DetermineNoticeTypeService.go('invitations')
+
+          expect(result.referenceCode).to.include('RINV-')
+          expect(result.referenceCode.length).to.equal(11)
+        })
+      })
+    })
+
+    describe('when the "notificationType" is "reminders"', () => {
+      it('creates a new session record', () => {
+        const result = DetermineNoticeTypeService.go('reminders')
+
+        expect(result).to.equal({
+          journey: 'reminders',
+          name: 'Returns: reminder',
+          notificationType: 'Returns reminder',
+          redirectPath: 'returns-period',
+          referenceCode: result.referenceCode, // randomly generated
+          subType: 'returnReminder'
+        })
+      })
+
+      describe('the "referenceCode" property', () => {
+        it('returns a reference code for "reminders" notifications', () => {
+          const result = DetermineNoticeTypeService.go('reminders')
+
+          expect(result.referenceCode).to.include('RREM-')
+          expect(result.referenceCode.length).to.equal(11)
+        })
+      })
+    })
+
+    describe('when the "notificationType" is "ad-hoc"', () => {
```

Comment:
```suggestion
    describe('and the "notificationType" is "ad-hoc"', () => {
```

Same question as before? Is this still here because this is a first step in refactoring ad-hoc out?

Reply:
Yes